packages/cli/src/commands/dev/dev.ts

```diff
@@ -186,30 +186,34 @@ export async function dev({
 
   const bundler = new DevBundler(env);
   bundler.__setLogger(logger);
-  await bundler.prepare(dotMastraPath);
-
-  const watcher = await bundler.watch(entryFile, dotMastraPath, discoveredTools);
-
-  const loadedEnv = await bundler.loadEnvVars();
-
-  // spread loadedEnv into process.env
-  for (const [key, value] of loadedEnv.entries()) {
-    process.env[key] = value;
-  }
 
+  // Get the port to use before prepare to set environment variables
   const serverOptions = await getServerOptions(entryFile, join(dotMastraPath, 'output'));
-
   let portToUse = port ?? serverOptions?.port ?? process.env.PORT;
   if (!portToUse || isNaN(Number(portToUse))) {
     const portList = Array.from({ length: 21 }, (_, i) => 4111 + i);
-
     portToUse = String(
       await getPort({
         port: portList,
       }),
     );
   }
 
+  // Set environment variables for playground development
+  process.env.MASTRA_DEV_PORT = portToUse.toString();
+  process.env.MASTRA_DEV_HOST = serverOptions?.host ?? 'localhost';
```

Comment:
This doesn't do anything as the playground is prebundled. Also let's not add global env vars here, we spawn a node server which should be used instead

```suggestion
```

Reply:
fixed it!